compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -1588,4 +1588,206 @@ bool isFullSlice(OffsetSizeAndStrideOpInterface sliceLoadStoreOp,
       sliceLoadStoreOp.getMixedStrides(), tensorType, dynamicDims);
 }
 
+//===----------------------------------------------------------------------===//
+// Utility functions for vector size inference for dynamic shapes
+//===----------------------------------------------------------------------===//
+
+std::optional<VectorizationTileSizes>
+inferSizesFromIR(linalg::LinalgOp linalgOp, std::optional<OpResult> opResult) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "Inferring sizes for:\n" << linalgOp;
+    if (opResult) {
+      llvm::dbgs() << " with OpResult.resultNumber="
+                   << opResult->getResultNumber();
+    }
+    llvm::dbgs() << '\n';
+  });
+
+  std::optional<vector::VscaleRange> vscaleRange;
+  if (!opResult) {
+    // Note: Inferring scalable sizes is not supported is `opResult` is set
+    // (which is used to compute sizes for tensor.pack/unpack).
+    auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(linalgOp);
+    vscaleRange = getDefaultVscaleRange(targetAttr);
```

Comment:
CC @banach-space who used to work on it. There is a request that needs to reuse the bounded shape inference, so they are moved to `Codegen/Utils`.

Reply:
Thanks for the ping! IIUC, this is just moving code around?

Reply:
yes.